packages/core/core-flows/src/definition/cart/workflows/refresh-payment-collection.ts

```diff
@@ -44,47 +23,66 @@ export const refreshPaymentCollectionForCartWorkflow = createWorkflow(
       entry_point: "cart",
       fields: [
         "id",
-        "total",
+        "region_id",
         "currency_code",
+        "total",
+        "raw_total",
         "payment_collection.id",
+        "payment_collection.raw_amount",
+        "payment_collection.amount",
+        "payment_collection.currency_code",
         "payment_collection.payment_sessions.id",
       ],
       variables: { id: input.cart_id },
       throw_if_key_not_found: true,
       list: false,
     })
 
-    const deletePaymentSessionInput = transform(
-      { paymentCollection: cart.payment_collection },
-      (data) => {
-        return {
-          ids:
-            data.paymentCollection?.payment_sessions
-              ?.map((ps) => ps.id)
-              ?.flat(1) || [],
-        }
-      }
-    )
+    when({ cart }, ({ cart }) => {
+      const valueIsEqual = MathBN.eq(
+        cart.payment_collection?.raw_amount ?? -1,
+        cart.raw_total
+      )
 
-    const updatePaymentCollectionInput = transform({ cart }, (data) => {
-      if (!isPresent(data.cart?.payment_collection?.id)) {
-        return
+      if (valueIsEqual) {
+        return cart.payment_collection.currency_code !== cart.currency_code
       }
 
-      return {
-        selector: { id: data.cart.payment_collection.id },
-        update: {
-          amount: data.cart.total,
-          currency_code: data.cart.currency_code,
-        },
-      }
-    })
+      return true
+    }).then(() => {
+      const deletePaymentSessionInput = transform(
```

Comment:
q: when payment value is equal, but payment session's payment provider is different, this wouldn't delete existing sessions right? That was an issue we addressed with this one

Reply:
From what I can understand, the only way to select a payment provider is by calling `/store/payment-collections/:id/payment-session` and that will trigger `createPaymentSessionsWorkflow` which will delete existing payment sessions.